ros/src/computing/planning/decision/packages/decision_maker/nodes/decision_maker/decision_maker_node_stateupdate.cpp

```diff
@@ -336,7 +336,20 @@ void DecisionMakerNode::updateStateStop(int status)
       layer_msg.data = "detectionarea";
       Pubs["filtering_gridmap_layer"].publish(layer_msg);
     }
-    if (current_velocity_ == 0.0 && !foundOtherVehicleForIntersectionStop_ && !timerflag)
+
+    // obtain distance from self to stopline -> TODO: refactor!
+    const static double STOPING_AREA_DISTANCE = 8.0;
+    geometry_msgs::Point pcw =
+      current_finalwaypoints_.waypoints[closest_waypoint_].pose.pose.position;
+    geometry_msgs::Point psw =
+      current_finalwaypoints_.waypoints[closest_stopline_waypoint_].pose.pose.position;
+    double distance_to_stopline =
+      std::hypot(std::hypot((pcw.x-psw.x), pcw.y-psw.y), pcw.z-psw.z);
+    bool inside_stopping_area = ((closest_waypoint_ < closest_stopline_waypoint_)
+      && distance_to_stopline < STOPING_AREA_DISTANCE);
+
+    if (current_velocity_ == 0.0 && !foundOtherVehicleForIntersectionStop_
```

Comment:
we see really lots of missed stoplines lately. Is it possible because we check `current_velocity_ == 0.0`? Please change it to compare against epsilon.

Reply:
fixed

Reply:
https://github.com/CPFL/Autoware/commit/a22e14eee0b769e63c79c023df0af9f18b7262be